packages/react-components/src/components/timegraph-output-component.tsx

```diff
@@ -384,7 +500,105 @@ export class TimegraphOutputComponent extends AbstractTreeOutputComponent<Timegr
         };
     }
 
+    private updateMarkersData(rangeEvents: TimelineChart.TimeGraphAnnotation[], newRange: TimelineChart.TimeGraphRange, newResolution: number) {
+        const annotationEntries: Entry[] = [];
+        const markers: Map<string, TimelineChart.TimeGraphState[]> = new Map();
+        const rows: TimelineChart.TimeGraphRowModel[] = [];
+        const filteredEvents = rangeEvents.filter(event => this.annotationMarkers?.includes(event.category));
+
+        filteredEvents.forEach(event => {
+            const categoryName = event.category;
+            if (!markers.has(categoryName)) {
+                markers.set(categoryName, []);
+            }
+            const states = markers.get(categoryName) || [];
+            const state = {
+                id: '1',
+                range: {
+                    start: event.range.start,
+                    end: event.range.end
+                },
+                label: event.label,
+                data: {
+                    style: event.data?.style,
+                    annotation: event
+                }
+            };
+            states.push(state);
+        });
+
+        const defaultRow = {
```

Comment:
I wish the default row would be smaller in height, but I don't think it's possible right now in timeline-chart...